### PR TITLE
fix: Ensure shutdown waits for pending promises

### PR DESCRIPTION
**Pull request by neilkakkar**
## Problem

Fixes https://github.com/PostHog/posthog/issues/13092

### Bump level

- [ ] Major
- [ ] Minor
- [x] Patch

### Libraries affected

- [ ] All of them
- [ ] posthog-web
- [x] posthog-node
- [ ] posthog-react-native

### Changelog notes

- Added support for X

### CODE REVIEW

File: posthog-node/test/posthog-node.spec.ts [hunk 1]
```diff
@@ -152,6 +152,63 @@ describe('PostHog Node.js', () => {
...
+            }),
+        } as any)
+      })
+    })
+
+    afterEach(() => {
+      posthog.debug(false)
+    })
+
+    it('should shutdown cleanly', async () => {
```
Comment (EDsCODE) [inline posthog-node/test/posthog-node.spec.ts:183]: very nice

**Review by EDsCODE — APPROVED** [review]

**Review by benjackwhite — APPROVED** [review]
Nice! Wanted to add this for ages

File: posthog-node/test/posthog-node.spec.ts [hunk 2]
```diff
@@ -152,6 +152,63 @@ describe('PostHog Node.js', () => {
     })
   })
 
+  describe('shutdown', () => {
```
Comment (benjackwhite) [inline posthog-node/test/posthog-node.spec.ts:155]: Although is this test meant to be in Node rather than core?

Comment (neilkakkar) [reply posthog-node/test/posthog-node.spec.ts:155]: yes 🙈

Comment (neilkakkar) [reply posthog-node/test/posthog-node.spec.ts:155]: oh, it's only core tests there, where shutdownAsync doesn't make too much sense 😓 . I'll keep it here for now! (vs. actually setting up coreStateless tests 🙈 )

**Pull request merged**